kubernetes-client/src/com/goyeau/kubernetes/client/KubernetesClient.scala

```diff
@@ -61,6 +63,24 @@ class KubernetesClient[F[_]: Async: Logger](
       encoder: Encoder[CustomResource[A, B]],
       decoder: Decoder[CustomResource[A, B]]
   ) = new CustomResourcesApi[F, A, B](httpClient, config, authorization, context)
+
+  def customRequest(
+      request: Request[F]
+  ): F[Request[F]] =
+    Request[F](
+      method = request.method,
+      uri = config.server.resolve(request.uri),
+      httpVersion = request.httpVersion,
+      headers = request.headers,
+      body = request.body,
+      attributes = request.attributes
+    ).withOptionalAuthorization(authorization)
```

Comment:
Why not using a `copy` here like below?

Reply:
http4s' `Request` is a `final class` (not `case class`) thus it doesn't provide the `.copy` methods unfortunately.